### PR TITLE
Add Button-like Link components

### PR DESCRIPTION
**Pull request by vitokhangnguyen**

### CODE REVIEW

File: src/ui/Button/Link.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import {
+  outlineStyle,
+  ghostStyle,
+  secondaryStyle,
+  primaryStyle,
+  linkStyle,
+  BaseProps,
+} from './styles';
+
+type LinkProps = Omit<HTMLProps<HTMLAnchorElement>, keyof BaseProps> &
```
Comment (Andrewnt219) [inline src/ui/Button/Link.tsx:12]: This looks complex. Try this. 
```tsx
type Props = NextLinkProps | AnchorProps

const isNextLinkProps = (props: Props): Props is NextLinkProps => {return true;}

() => {
if(isNextLinkProps) return <NextLink />
return <a />
}
```

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]
Wonderful, just one typing improvement.

File: src/ui/Button/Link.tsx (hunk 1 above)
Comment (vitokhangnguyen) [reply src/ui/Button/Link.tsx:12]: The `Omit<HTMLProps<HTMLAnchorElement>, keyof BaseProps>` is to solve a werid TypeScript issue.

The `HTMLProps<HTMLAnchorElement>` (`AnchorProps`) type somehow has a `size` property in it which is not supposed to be there. When unioned with `BaseProps`, the 2 `size` properties conflict and result in the type `undefined`.

That's why I exclude all `keyof BaseProps` from the `AnchorProps` although they're tehcnically not there in the first place.

Comment (vitokhangnguyen) [reply src/ui/Button/Link.tsx:12]: And sometime we might need props in both NextLink and <a> so I don't think this would do...

Comment (Andrewnt219) [reply src/ui/Button/Link.tsx:12]: NextLinkProps = LinkProps & AnchorProps;

Comment (vitokhangnguyen) [reply src/ui/Button/Link.tsx:12]: I'll try that but ate you sure they don't have any overlapping props besides href?

**Review by Andrewnt219 — APPROVED** [review]

**Pull request merged**